.circleci/config.yml

```diff
@@ -0,0 +1,321 @@
+            sudo cp /tmp/dl/swagger /usr/local/bin/swagger
+            sudo chmod +x /usr/local/bin/swagger
+            swagger version
+      - run:
+          name: Install Ksonnet v0.13.1
+          command: |
+            set -x
+            [ -e /tmp/dl/ks.tar.gz ] || curl -sLf -C - -o /tmp/dl/ks.tar.gz https://github.com/ksonnet/ksonnet/releases/download/v0.13.1/ks_0.13.1_linux_amd64.tar.gz
+            tar -C /tmp -xf /tmp/dl/ks.tar.gz
+            sudo cp /tmp/ks_0.13.1_linux_amd64/ks /usr/local/go/bin/ks
+            sudo chmod +x /usr/local/go/bin/ks
+            ks version
+      - run:
+          name: Install Helm v2.13.1
+          command: |
+            set -x
+            [ -e /tmp/dl/helm.tar.gz ] || curl -sLf -C - -o /tmp/dl/helm.tar.gz https://storage.googleapis.com/kubernetes-helm/helm-v2.13.1-linux-amd64.tar.gz
+            tar -C /tmp/ -xf /tmp/dl/helm.tar.gz
+            sudo cp /tmp/linux-amd64/helm /usr/local/go/bin/helm
+            helm version --client
+            helm init --client-only
+      - run:
+          name: Install Kustomize v1.0.11
+          command: |
+            set -x
+            [ -e /tmp/dl/kustomize1 ] || curl -sLf -C - -o /tmp/dl/kustomize1 https://github.com/kubernetes-sigs/kustomize/releases/download/v1.0.11/kustomize_1.0.11_linux_amd64
+            sudo cp /tmp/dl/kustomize1 /usr/local/go/bin/
+            sudo chmod +x /usr/local/go/bin/kustomize1
+            kustomize1 version
+      - run:
+          name: Install Kustomize v2.0.3
+          command: |
+            set -x
+            [ -e /tmp/dl/kustomize ] || curl -sLf -C - -o /tmp/dl/kustomize https://github.com/kubernetes-sigs/kustomize/releases/download/v2.0.3/kustomize_2.0.3_linux_amd64
+            sudo cp /tmp/dl/kustomize /usr/local/go/bin/
+            sudo chmod +x /usr/local/go/bin/kustomize
+            kustomize version
+      - run:
+          name: Install Protobuf compiler v3.7.1
+          command: |
+            set -x
+            [ -e /tmp/dl/protoc.zip ] || curl -sLf -C - -o /tmp/dl/protoc.zip https://github.com/protocolbuffers/protobuf/releases/download/v3.7.1/protoc-3.7.1-linux-x86_64.zip
+            sudo unzip /tmp/dl/protoc.zip bin/protoc -d /usr/local/
+            sudo chmod +x /usr/local/bin/protoc
+            sudo unzip /tmp/dl/protoc.zip include/* -d /usr/local/
+            protoc --version
+      - save_cache:
+          key: dl-v4
+          paths:
+            - /tmp/dl
+  save_go_cache:
+    steps:
+      - save_cache:
+          key: go-v14-{{ .Branch }}
+          paths:
+            - /home/circleci/.go_workspace
+            - /home/circleci/.cache/go-build
+            - /home/circleci/sdk/go1.11.4
+  restore_go_cache:
+    steps:
+      - restore_cache:
+          keys:
+            - go-v14-{{ .Branch }}
+            - go-v14-master
+jobs:
+  test:
+    working_directory: /home/circleci/.go_workspace/src/github.com/argoproj/argo-cd
+    machine:
+      image: circleci/classic:201808-01
```

Comment:
Is it possible to use `argoproj/argo-cd-ci-builder`? I guess you've tried already. Can you share why it did not work?

Reply:
You can't use that image if you're using the `machine` executor. I needed to use that executor so that I can run `k3s`. It is possible, in a v2, we could move to the Docker executor and then use the `k3s` docker image, but I'd like to get this soaked on a number of PRs first.

Reply:
Note - I think that the docker executor is slightly faster (15s) and more reliable than the machine executor. I don't feel like Circle CI give much TLC to the machine executor.